### PR TITLE
#1317 - Ensure exchange order tables have minimum widths

### PR DESCRIPTION
**Pull request by calvinfroedge**
Set minimum widths for tables so that data will render correctly even on responsive / condensed views (with scrolling):

update

### CODE REVIEW

**Comment by svk31** [comment]
This makes it so that there are permanent scrollbars around all the boxes in the exchange:

screenshot from 2018-03-30 10-16-26

It also doesn't use the perfect-scrollbar scrollbars that the existing vertical scrollbars use. If you're on macOS you probably never noticed because the native scrollbars there look the same as the perfect-scrollbar bars, but on Windows or Linux you'll get clashing scrollbars like you see in my screenshot.

File: app/assets/stylesheets/components/_exchange.scss [hunk 1]
```diff
@@ -581,6 +581,32 @@ table thead tr th.mymarkets-header {
...
+    overflow: scroll;
+}
+
+.orderbook-table {
+    min-width: 360px;
+}
+
+.orderbook-table.buyorders td:first-child,
+.orderbook-table.buyorders th:first-child {
+    width: 100px;
```
Comment (svk31) [inline app/assets/stylesheets/components/_exchange.scss]: I'm not a fan of fixed widths, please use a min-width instead.

File: app/assets/stylesheets/components/_exchange.scss [hunk 2]
```diff
@@ -581,6 +581,32 @@ table thead tr th.mymarkets-header {
...
+}
+
+.orderbook-table.tradehistory,
+.orderbook-table.openorders {
+    min-width: 300px;
+}
+
+.orderbook-table.tradehistory td:last-child,
+.orderbook-table.tradehistory th:last-child {
+    width: 120px;
```
Comment (svk31) [inline app/assets/stylesheets/components/_exchange.scss]: Same as above re fixed widths.

**Comment by calvinfroedge** [comment]
Will need some additional time to use PS to handle this...also min-width was not supported based on the table display, which is why I used fixed width.

**Comment by calvinfroedge** [comment]
@svk31 PR has been updated so that perfect scrollbar is used on overages. One big change to note is that to do this I had to keep the headers fixed, so the data does not stay aligned with the headers. I can correct this, but then headers will not be fixed when doing vertical scrolling.

I set the point at which horizontal scroll gets kicked in at 350px...so 350px is basically the minimum width for all tables. This seems to correct most problems with huge numbers in columns, but there are still some columns with very long values that bleed over. In those cases I might suggest that we use ellipses after a certain width and then show the full value in a mouseover. Open to feedback.

update

**Comment by abitmore** [comment]
>  One big change to note is that to do this I had to keep the headers fixed, so the data does not stay aligned with the headers. I can correct this, but then headers will not be fixed when doing vertical scrolling.

Use an invisible horizontal scroll bar for the header, when the table below scrolls horizontally, automatically scroll the header as well?

**Comment by calvinfroedge** [comment]
@abitmore I don't think it's possible to accomplish this with CSS + perfect scrollbar as making them scroll together requires putting the table content and the table headers in the same container. This will result in the headers not being fixed anymore. So..it's kind of take your pick...fixed headers and table content that scrolls independently or headers that scroll with the table content (but are no longer fixed).

**Comment by abitmore** [comment]
Understood. But to me it's strange that the header has fixed width but the data rows are wider. I'm not the one who makes decision here though.

**Comment by calvinfroedge** [comment]
@svk31 Can you please chime in here

**Comment by calvinfroedge** [comment]
@svk31 I'm installing a linux VM so that hopefully I can see the same thing you are.

**Comment by sschiessl-bcp** [comment]
> I'm installing a linux VM so that hopefully I can see the same thing you are.

Were you able to replicate this?

**Comment by svk31** [comment]
Closing this since it's too old by now, wasn't working properly and being superseded by the exchange redesign anyway.

**Pull request closed without merging**